File: java/hamming/src/main/java/Hamming.java
public class Hamming {

  public static int compute(String strand1, String strand2) {
    if (strand1.length() != strand2.length()) {
      throw new IllegalArgumentException("stands not equals");
    }

    int distance = 0;

    for (int i = 0; i < strand1.length(); ++i) {
      if (strand1.charAt(i) != strand2.charAt(i)) {
        distance += 1;
      }
    }

    return distance;
  }

}

File: lua/hamming/hamming.lua
local M = {}

local char_at = function(text, i) return text:sub(i, i) end

M.compute = function(strand1, strand2)
  local distance = 0
  for i = 1, #strand1 do
    if char_at(strand1, i) ~= char_at(strand2, i) then
      distance = distance + 1
    end
  end
  return distance 
end

return M